### ISSUE TITLE
condition-wrapping should behave the same way for all operators

### ISSUE THREAD

**Issue by mklnwt** [pinterest/ktlint#3000]
Parentheses should behave the same way for all operators in `condition-wrapping`.

For mathematical operator ktlint behaves correctly.
```kotlin
fun foo(
    a: Int,
    b: Int,
    c: Int,
): Int =
    a + (
        b - c
    )
```

If we add them for Booleans it works the same way.
```kotlin
fun bar(
    a: Boolean,
    b: Boolean,
    c: Boolean,
): Boolean =
    a * (
        b + c
    )

operator fun Boolean.times(other: Boolean): Boolean = this && other

operator fun Boolean.plus(other: Boolean): Boolean = this || other
```

## Expected Behavior

This should work the same way for other operators.
```kotlin
fun bar(
    a: Boolean,
    b: Boolean,
    c: Boolean,
): Boolean =
    a && (
        b || c
    )
```

## Current Behavior

Instead parentheses are broken into an empty line.
```kotlin
fun bar(
    a: Boolean,
    b: Boolean,
    c: Boolean,
): Boolean =
    a &&
        (
            b || c
        )
```

## Additional information
* Current version of ktlint: 1.6.0
```
[*.{kt,kts}]
max_line_length = 120
ktlint_code_style = ktlint_official
```

**Comment by paul-dingemans** [comment]
Thanks for spotting and reporting this inconsistency. I rather would say the arithmetic multiline expression is wrapped wrongly instead of the logical expression. In `ktlint_official` code style it would be more fitting to write:
```
fun foo(
    a: Int,
    b: Int,
    c: Int,
): Int =
    a + 
        (
            b - c
        )
```
or
```
fun foo(
    a: Int,
    b: Int,
    c: Int,
): Int =
    a + 
        (b - c)
```
The latter one should be preferred if the sub expression between parenthesis fit on a single line.

**Comment by mklnwt** [comment]
I would suggest 
``` kotlin
fun foo(
    a: Int,
    b: Int,
    c: Int,
): Int =
    a + (
        b - c
    )
```
or
```kotlin
fun foo(
    a: Int,
    b: Int,
    c: Int,
): Int =
    a + 
        (b - c)
```
or
```kotlin
fun foo(
    a: Int,
    b: Int,
    c: Int,
): Int =
    a + (b - c)
```

I think the latter 2 are already accepted by the formatter.
For small arithmetics it shouldn't be required to break into multiple lines.

For the former, if `b` & `c` are a bit longer `(b - c)` might not be an option.
But I don't see the point of keeping `(` in a line alone.
The same way as `b - c` can be in one line, `a + (` should be allowed in one line.

Additionally it would be more in line with this as well
``` kotlin
fun foo(
    a: Int,
    b: Int,
    c: Int,
    d: Int,
): Int =
    (a - d) +
        (b - c)
```

**Comment by paul-dingemans** [comment]
With simple names like `a`, `b`, `c`, and `d` it all looks simple and elegant. However when you experiment with longer names, you will need to deal with more complications. Also, you need to take into account the other constructs like `if`, `while` etc.

In example below we have some longer variable names:
```
// Max line length --------------------------------------------------------------------------------
fun bar3(
    a: Boolean,
    someLongConditions: Boolean,
    someOtherLonggggggggggggggggggggggggggggggggggggggggCondition: Boolean,
): Boolean {
    if (a &&
        (someLongConditions || someOtherLonggggggggggggggggggggggggggggggggggggggggCondition)
    ) {
        // ..
    }
    while (a &&
        (someLongConditions || someOtherLonggggggggggggggggggggggggggggggggggggggggCondition)
    ) {
        // ..
    }
    return a &&
        (someLongConditions || someOtherLonggggggggggggggggggggggggggggggggggggggggCondition)
}
```
where below just does not fit on a single line:
```
if (a && (someLongConditions || someOtherLonggggggggggggggggggggggggggggggggggggggggCondition)) {
```

When the variable `someLongConditions` is made longer it eventually leads to the situation that it does not fit on the same line with `someOtherLonggggggggggggggggggggggggggggggggggggggggCondition`. The formatting below, is still readable and consistent:
```
// Max line length --------------------------------------------------------------------------------
fun bar3(
    a: Boolean,
    someLonggggggggggggggggggggggggggggggggggggggggConditions: Boolean,
    someOtherLonggggggggggggggggggggggggggggggggggggggggCondition: Boolean,
): Boolean {
    if (a &&
        (
            someLonggggggggggggggggggggggggggggggggggggggggConditions ||
                someOtherLonggggggggggggggggggggggggggggggggggggggggCondition
        )
    ) {
        // ..
    }
    while (a &&
        (
            someLonggggggggggggggggggggggggggggggggggggggggConditions ||
                someOtherLonggggggggggggggggggggggggggggggggggggggggCondition
        )
    ) {
        // ..
    }
    return a &&
        (
            someLonggggggggggggggggggggggggggggggggggggggggConditions ||
                someOtherLonggggggggggggggggggggggggggggggggggggggggCondition
        )
}
```

When allowing the opening parenthesis on the same line, the code sample above would look like below:
```
fun bar2(
    a: Boolean,
    someLonggggggggggggggggggggggggggggggggggggggggConditions: Boolean,
    someOtherLonggggggggggggggggggggggggggggggggggggggggCondition: Boolean,
): Boolean {
    if (a && (
        someLonggggggggggggggggggggggggggggggggggggggggConditions ||
            someOtherLonggggggggggggggggggggggggggggggggggggggggCondition
    )
    ) {
        // ..
    }
    while (a && (
        someLonggggggggggggggggggggggggggggggggggggggggConditions ||
            someOtherLonggggggggggggggggggggggggggggggggggggggggCondition
    )
    ) {
        // ..
    }
    return a && (
        someLonggggggggggggggggggggggggggggggggggggggggConditions ||
            someOtherLonggggggggggggggggggggggggggggggggggggggggCondition
    )
}
```

**Comment by mklnwt** [comment]
What's the issue with formatting this
```kotlin
// Max line length --------------------------------------------------------------------------------
fun bar3(
    a: Boolean,
    someLonggggggggggggggggggggggggggggggggggggggggConditions: Boolean,
    someOtherLonggggggggggggggggggggggggggggggggggggggggCondition: Boolean,
): Boolean {
    if (a &&
        (
            someLonggggggggggggggggggggggggggggggggggggggggConditions ||
                someOtherLonggggggggggggggggggggggggggggggggggggggggCondition
        )
    ) {
        // ..
    }
    while (a &&
        (
            someLonggggggggggggggggggggggggggggggggggggggggConditions ||
                someOtherLonggggggggggggggggggggggggggggggggggggggggCondition
        )
    ) {
        // ..
    }
    return a &&
        (
            someLonggggggggggggggggggggggggggggggggggggggggConditions ||
                someOtherLonggggggggggggggggggggggggggggggggggggggggCondition
        )
}
```

like this instead?
```kotlin
// Max line length --------------------------------------------------------------------------------
fun bar3(
    a: Boolean,
    someLonggggggggggggggggggggggggggggggggggggggggConditions: Boolean,
    someOtherLonggggggggggggggggggggggggggggggggggggggggCondition: Boolean,
): Boolean {
    if (
        a && (
            someLonggggggggggggggggggggggggggggggggggggggggConditions ||
                someOtherLonggggggggggggggggggggggggggggggggggggggggCondition
        )
    ) {
        // ..
    }
    while (
        a && (
            someLonggggggggggggggggggggggggggggggggggggggggConditions ||
                someOtherLonggggggggggggggggggggggggggggggggggggggggCondition
        )
    ) {
        // ..
    }
    return a && (
        someLonggggggggggggggggggggggggggggggggggggggggConditions ||
            someOtherLonggggggggggggggggggggggggggggggggggggggggCondition
    )
}
```

I think one of the checks doesn't allow `if (a &&` anyway, so it would be broken down into a new line.
But the 3rd example (`return a &&`) illustrates this pretty well. It doesn't really matter how long the variable names become.
Even if you make `a` a lot longer, having the `(` on the same line doesn't seem to change the readability since the operator (i.e. `&&`) is already on the same line.

**Comment by paul-dingemans** [comment]
Inherently there is nothing wrong with one format or the other. The format proposed by me just better aligns what Ktlint is currently producing. So I will make it consistent with the current formatting of arithmetic expressions.

**Comment by paul-dingemans** [comment]
Arithmentic and logic expression operands are now wrapped in same way by new rule `expression-operand-wrapping` rule.

Format below is still not accepted, and now is also no longer allowed on arithmetic expressions:
```
fun bar(
    a: Boolean,
    b: Boolean,
    c: Boolean,
): Boolean =
    a && (
        b || c
    )
```
